### PR TITLE
#169836672 Create migrations

### PR DESCRIPTION
**Pull request by EMacco**
#### What does this PR do?
This PR creates the migrations for the various tables 

#### Description of Task to be completed?
- create migrations for all tables
- create models and associations

#### How should this be manually tested?
- Checkout to this branch 
- Run the command `yarn install`
- Create database called `abarian_db`
- In your `.env` file, add the following fields `DB_USERNAME=`username, `DB_DATABASE=abarian_db`, `DB_HOST=localhost`, `DB_PASSWORD=`password_if_applicable
- Run the command `yarn run migrate`
- All tables should be created. You can confirm using Postico

#### Any background context you want to provide?
- You need to have downloaded and setup `Postgresql` on your local machine

#### What are the relevant pivotal tracker stories?
[#169836672](https://www.pivotaltracker.com/story/show/169836672)

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A

### CODE REVIEW

File: server/migrations/001-add-extension.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const isTestEnvironment = process.env.NODE_ENV === "test";
+
+export const up = (queryInterface) => {
+  if (isTestEnvironment) {
+    return new Promise((resolve) => resolve(1));
+  }
+  return queryInterface.sequelize.query(
+    "CREATE EXTENSION IF NOT EXISTS citext WITH SCHEMA public;"
+  );
```
[the remark this answers is not in the document]
Comment (kachikelz) [reply server/migrations/001-add-extension.js]: Does the hound suggestion imply any error or just a coding standerd

File: server/migrations/20191122150954-create-user.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+module.exports = {
```
Comment (kachikelz) [inline server/migrations/20191122150954-create-user.js]: Do u think maybr the user needs an [ isValid ] boolean table to show if user is a valid user or not.

File: server/models/index.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+const db = {};
+
+let sequelize;
+if (config.use_env_variable) {
+  sequelize = new Sequelize(process.env[config.use_env_variable], config);
+} else {
+  const {
+    database, username, password, ...otherConfig
+  } = config;
+  sequelize = new Sequelize(database, username, password, {
```
Comment (kachikelz) [inline server/models/index.js]: pg returns a promise and a failure to connect to the db cud have a .catch to return log any error ?

File: server/models/order.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      paid: {
+        type: DataTypes.BOOLEAN,
+        allowNull: false
+      },
+      delivery: {
+        type: DataTypes.STRING,
+        allowNull: false
+      }
+    },
+    {}
```
Comment (kachikelz) [inline server/models/order.js]: Is the empty bracket serving a purpose here?

File: server/migrations/20191122150954-create-user.js (hunk 2 above)
Comment (EMacco) [reply server/migrations/20191122150954-create-user.js]: Yes that’s why I have a field called active. It’ll become true when the user account is verified

File: server/migrations/001-add-extension.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (EMacco) [reply server/migrations/001-add-extension.js]: Coding standard.

**Pull request merged**